lib/instream/connection/query_runner_v2.ex

```diff
@@ -144,6 +145,34 @@ defmodule Instream.Connection.QueryRunnerV2 do
     result
   end
 
+  @doc """
+  Executes `:delete` predicate.
+  """
+  @spec delete(map(), Keyword.t(), module) :: any
+  def delete(points, opts, conn) do
+    config = conn.config()
+
+    {query_time, result} =
+      :timer.tc(fn ->
+        points
+        |> config[:deleter].delete(opts, conn)
```

Comment:
Would it be reasonable to inline the code for the delete request?

The `writer` (I assume the structure here is coming from there) has a process/config structure to support the UDP writer with the long running UDP socket. The `deleter` however only sends a single request directly via the HTTP client.

If the need arises to change what is done on a delete request it could always be added but for the time being it seems to be unnecessary complex.

Reply:
Removed calculating `query_time`.

Reply:
I did not mean that the `query_time` should be removed but whether it is reasonable to inline everything the `Instream.Deleter.Predicate` does.

Something like this (with the logging/result return also to be added):

```elixir
  @spec delete(Instream.Connection.delete_request(), Keyword.t(), module) :: any
  def delete(payload, opts, conn) do
    config = conn.config()
    headers = Headers.assemble(config, opts) ++ [{"Content-Type", "application/json"}]
    http_opts = http_opts(config, opts)
    body = JSON.encode(payload, conn)
    url = URL.delete(config, opts)

    {query_time, result} =
      :timer.tc(fn ->
        config[:http_client].request(:post, url, headers, body, http_opts)
      end)

    # ...
  end
```

The current structure just seems far too complex for what it tries to achieve while still missing some of the things it suggests it should achieve. For example:

- the logging adds a `"predicate"` to the message, that would be wrong if the `:deleter` changes
- there is a `deleter_workers` callback but the workers are never started
- the interface of `Instream.Connection.delete/2` may prevent any sensible different deleter implementation because the type is completely fixed to predicate deletion

Reply:
Oh, I get it now. You are right. Done.